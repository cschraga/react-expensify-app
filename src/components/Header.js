import React   from 'react';
import {NavLink} from 'react-router-dom';

const Header= () => (
    <div>
        <h1>Welcome To Expensify</h1>
        <h3><NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink> &nbsp;
            <NavLink to="/create" activeClassName="is-active">Create</NavLink> &nbsp;
            <NavLink to="/help" activeClassName="is-active">Help</NavLink>
        </h3>
    </div>
);

export default Header;