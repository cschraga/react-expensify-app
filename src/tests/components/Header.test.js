import React from 'react';
//import ReactShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';
//import toJSON from 'enzyme-to-json'; //dont need anymore now that we added snapshot serializer
import Header from '../../components/Header';

test('should render header correctly', ()=>{
    //using enzyme
    const wrapper = shallow(<Header />);
    //expect(wrapper.find('h1').text()).toBe("Welcome To Expensify");
    expect(wrapper).toMatchSnapshot();
    //React Shallow Renderer way ... not as good
    //const renderer = new ReactShallowRenderer();
    //renderer.render(<Header />);
    //expect(renderer.getRenderOutput()).toMatchSnapshot();
    //console.log(renderer.getRenderOutput());
});

