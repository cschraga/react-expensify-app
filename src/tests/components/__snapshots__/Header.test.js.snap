// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<div>
  <h1>
    Welcome To Expensify
  </h1>
  <h3>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      exact={true}
      to="/"
    >
      Home
    </NavLink>
      
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      to="/create"
    >
      Create
    </NavLink>
      
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      to="/help"
    >
      Help
    </NavLink>
  </h3>
</div>
`;
